### PR TITLE
vim-patch:8.1.{2342,2343,2356,2357},8.2.0233: `rand` and `srand`

### PR DESCRIPTION
**Pull request by seandewar**

### CODE REVIEW

**Comment by zeertzjq** [comment]
I resolved the merge conflicts.

File: src/nvim/testdir/test_random.vim [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  " let s = srand()
+  " if !has('win32') && filereadable('/dev/urandom')
+  "   " using /dev/urandom
+  "   call assert_notequal(s, srand())
+  " else
+  "   " using time()
+  "   call assert_equal(s, srand())
+  "   call test_settime(12341235)
+  "   call assert_notequal(s, srand())
+  " endif
```
Comment (zeertzjq) [inline src/nvim/testdir/test_random.vim:23]: Does only commenting out the part using `test_settime()` work?
```suggestion
  if !has('win32') && filereadable('/dev/urandom')
    " using /dev/urandom
    call assert_notequal(s, srand())
  " else
  "   " using time()
  "   call assert_equal(s, srand())
  "   call test_settime(12341235)
  "   call assert_notequal(s, srand())
  endif
```

Comment (zeertzjq) [reply src/nvim/testdir/test_random.vim:23]: Oops, ignore this comment.

Comment (zeertzjq) [inline src/nvim/testdir/test_random.vim:23]: I think uncommenting the part that does not use `test_settime()` should be fine?
```suggestion
  let s = srand()
  if !has('win32') && filereadable('/dev/urandom')
    " using /dev/urandom
    call assert_notequal(s, srand())
  " else
  "   " using time()
  "   call assert_equal(s, srand())
  "   call test_settime(12341235)
  "   call assert_notequal(s, srand())
  endif
```

**Review by zeertzjq — APPROVED** [review]

File: src/nvim/testdir/test_random.vim (hunk 1 above)
Comment (seandewar) [reply src/nvim/testdir/test_random.vim:23]: I'd imagine there's a very, _very_ small chance they can be equal without the `test_settime` call above?
But it's so unlikely that it probably doesn't matter 😆

Comment (zeertzjq) [reply src/nvim/testdir/test_random.vim]: It is also the case in Vim.

Comment (zeertzjq) [reply src/nvim/testdir/test_random.vim]: I think `test_settime()` should not affect `/dev/urandom`?

Comment (seandewar) [reply src/nvim/testdir/test_random.vim]: Oh yeah, I totally forgot about that. 🤦 🤦 🤦

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**